logproof/src/linear_relation.rs

```diff
@@ -641,6 +700,30 @@ impl LogProof {
         (g * phi_inv).into_iter().collect()
     }
 
+    /**
+     * Uses the single elements in a first vector to scale a vector of vectors,
+     * and flattens the result. If the each vector happens to be the same
+     * length, then this is the same as the following operation:
+     * `vec(diag(v) * M)` where `vec` is row major vectorization.
+     *
+     * - v: Vector to diagonalize
+     * - m: Matrix as an array of vectors
+     *
+     * Note: the elements in `m` do not need to be the same size.
+     */
+    pub(crate) fn scale_rows_and_flatten<T>(v: &[T], m: &[Vec<T>]) -> Vec<T>
```

Comment:
It's not the most amazing name since each row can be a different length but it was the best I could come up with. 🤷